src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -2390,38 +2419,25 @@ internal bool ContainsNoPiaLocalTypes()
             return _lazyContainsNoPiaLocalTypes == ThreeState.True;
         }
 
-        internal bool HasNullableAttribute(EntityHandle token, out ImmutableArray<bool> nullableTransforms)
+        internal bool HasNullableAttribute(EntityHandle token, out byte defaultTransform, out ImmutableArray<byte> nullableTransforms)
         {
             AttributeInfo info = FindTargetAttribute(token, AttributeDescription.NullableAttribute);
             Debug.Assert(!info.HasValue || info.SignatureIndex == 0 || info.SignatureIndex == 1);
 
+            defaultTransform = 0;
```

Comment:
>0; [](start = 31, length = 2)

Consider using another value to represent "not a value" and only set to 0 at line 2437 (from `TryExtractValueFromAttribute`).

Reply:
> Consider using another value to represent "not a value" and only set to 0 at line 2437 (from TryExtractValueFromAttribute).

I see no benefit in inventing new "defaults" when this value just works, this is a well established pattern for functions that return information via out parameters and return boolean value indicating success of the operation. Consumers that really depend on the fact whether the attribute was found and cracked successfully should simply check return value of this method. 

---
In reply to: [234368109](https://github.com/dotnet/roslyn/pull/31212#discussion_r234368109) [](ancestors = 234368109)